### PR TITLE
RemoteData support for the Logs model

### PR DESCRIPTION
**Pull request by MauritsioRK**
https://alice.its.cern.ch/jira/browse/O2B-303

### CODE REVIEW

File: lib/public/views/Logs/Details/index.js [hunk 1]
```diff
@@ -48,15 +47,18 @@ const logDetailScreen = (model) => {
...
-    } else if (error) {
         return h('', [
-            h('.danger', 'This log could not be found.'),
-            h('button.btn.btn-primary.mv3', {
+            h('h2.mv2', { onremove: () => model.logs.clearLogs() }, 'Log Tree'),
+            h('.w-100.flex-column', tree(data.payload[0])),
+        ]);
+    } else if (data.isFailure()) {
+        return h('', [
+            h('.alert.alert-danger', 'This log could not be found.'),
```
Comment (mvegter) [inline lib/public/views/Logs/Details/index.js:56]: We are not sure what the errors is, so we should use something like:
```js
data.payload.map((error) => h('.alert.alert-danger', h('b', `${error.title}: `), error.detail))
```

File: lib/public/views/Logs/Logs.js [hunk 2]
```diff
@@ -24,18 +24,18 @@ export default class Overview extends Observable {
     constructor(model) {
         super();
         this.model = model;
+
         this.filterCriteria = [];
-        this.data = [];
-        this.filteredData = [];
+        this.logs = RemoteData.NotAsked();
```
Comment (mvegter) [inline lib/public/views/Logs/Logs.js:29]: You can replace this with the `clearLogs` method

File: lib/public/views/Logs/Logs.js [hunk 3]
```diff
@@ -82,91 +110,24 @@ export default class Overview extends Observable {
...
-            const logId = result.data.id;
+            this.createdLog = RemoteData.success(result.data);
             this.title = '';
             this.text = '';
-            this.error = false;
 
-            await this.fetchOneLog(logId);
-            await this.model.router.go(`/?page=entry&id=${logId}`);
+            await this.model.router.go(`/?page=entry&id=${this.createdLog.payload.id}`);
             this.notify();
```
Comment (mvegter) [inline lib/public/views/Logs/Logs.js:118]: Is this notify still needed if we are going to redirect?

File: lib/public/views/Logs/Logs.js [hunk 4]
```diff
@@ -334,4 +315,33 @@ export default class Overview extends Observable {
...
+    setText(text) {
+        this.text = text;
+        this.notify();
+    }
+
+    /**
+     * Sets all data related to the Logs to `NotAsked`.
+     * @returns {undefined}
+     */
+    clearLogs() {
```
Comment (mvegter) [inline lib/public/views/Logs/Logs.js:343]: If the page is destroyed, shouldn't we also reset the paginiation options here?

File: lib/public/views/Logs/Details/index.js (hunk 1 above)
Comment (MauritsioRK) [reply lib/public/views/Logs/Details/index.js]: Done

File: lib/public/views/Logs/Logs.js (hunk 2 above)
Comment (MauritsioRK) [reply lib/public/views/Logs/Logs.js]: Done

File: lib/public/views/Logs/Logs.js (hunk 3 above)
Comment (MauritsioRK) [reply lib/public/views/Logs/Logs.js]: I had a look, and it did not seem detrimental - removed

File: lib/public/views/Logs/Logs.js (hunk 4 above)
Comment (MauritsioRK) [reply lib/public/views/Logs/Logs.js:334]: Good point - done

**Pull request merged**